### DISCUSSION TITLE
Short description of features for SVGGraphics2D

### DISCUSSION THREAD

**Discussion by istinnstudio** [css4j/echosvg#66] [Q&A] [answered]
Hi, I am using batik to translate Graphics2D elements to SVG with the SVGGraphics2D Generator. I ve never tried CSS with an SVG before but I know the benefits in styling/animation. Does this project act as an additional level renderer if a CSS file will be provided? Something like a post processor through CSS? Could you provide a general short description of what we can possibly gain in front of the classic SVGGraphics2D Generator? 

Regards

**Comment by carlosame** [comment, marked as answer]
With `SVGGraphics2D` you aren't using CSS as input: EchoSVG's `SVGGraphics2D` is almost the same as Batik's one, except for a few improvements like #16 or #17. Those fixes are one of the reasons why EchoSVG passes all of the Batik's `svggen` unit tests while Batik itself doesn't.

This project has no specific documentation about `SVGGraphics2D` currently, except for the Javadocs which are linked from the Wiki.

**Reply by istinnstudio** [reply]
Thanks for the info

**Reply by carlosame** [reply]
In my reply I mentioned the Batik tests, and in another discussion I talked about reproducibility issues with Batik's test suite. To put some context about the differences between Batik and EchoSVG testing infrastructure I created a new Wiki page:

https://github.com/css4j/echosvg/wiki/From-Batik-test-suite-to-EchoSVG-tests

**Comment by carlosame** [comment]
Now there is a Wiki page about `SVGGraphics2D` which includes the new features introduced in 1.1, like the configurable compression level or the support for arbitrary color spaces:

https://github.com/css4j/echosvg/wiki/SVG-Generator
